### PR TITLE
Bugfix in TextBasedProtocolMessage::parseFields() (issue #176)

### PR DESCRIPTION
**Pull request by gx740**

### CODE REVIEW

File: Packet++/src/TextBasedProtocol.cpp [hunk 1]
```diff
@@ -112,13 +112,21 @@ void TextBasedProtocolMessage::parseFields()
...
 		curOffset += curField->getFieldSize();
 		HeaderField* newField = new HeaderField(this, curOffset, nameValueSeperator, spacesAllowedBetweenNameAndValue);
-		LOG_DEBUG("Added new field: name='%s'; offset in packet=%d; length=%d", newField->getFieldName().c_str(), newField->m_NameOffsetInMessage, (int)newField->getFieldSize());
-		LOG_DEBUG("     Field value = %s", newField->getFieldValue().c_str());
-		curField->setNextField(newField);
-		curField = curField->getNextField();
-		fieldName = newField->getFieldName();
-		std::transform(fieldName.begin(), fieldName.end(), fieldName.begin(), ::tolower);
-		m_FieldNameToFieldMap.insert(std::pair<std::string, HeaderField*>(fieldName, newField));
+		if(newField->getFieldSize() > 0)
```
Comment (gx740) [inline Packet++/src/TextBasedProtocol.cpp]: newField->getFieldSize() returns zero therefore loop never ends

**Comment by seladb** [comment]
Thanks for submitting this PR! I see it contains a few changes and fixes. Could you please separate them and submit each one as a separate PR? For example, this PR should only contain the bugfix to #176

**Comment by gx740** [comment]
It's my first PR therefore so many mistakes. I removed other commits.

File: Packet++/src/TextBasedProtocol.cpp [hunk 2]
```diff
@@ -112,13 +112,21 @@ void TextBasedProtocolMessage::parseFields()
...
+			LOG_DEBUG("     Field value = %s", newField->getFieldValue().c_str());
+			curField->setNextField(newField);
+			curField = newField;
+			fieldName = newField->getFieldName();
+			std::transform(fieldName.begin(), fieldName.end(), fieldName.begin(), ::tolower);
+			m_FieldNameToFieldMap.insert(std::pair<std::string, HeaderField *>(fieldName, newField));
+		}
+		else
+		{
+			delete newField;
```
Comment (gx740) [inline Packet++/src/TextBasedProtocol.cpp]: The method isHeaderComplete will return false value for this layer.

**Comment by seladb** [comment]
Do you have a pcap example that can reproduce this issue?
It'd be good if we can add this to the unit-tests. Please send me the pcap file if you have it, I can add the tests

**Comment by gx740** [comment]
I already attached this file to the issue #176 (dump.zip).

**Pull request merged**